apps/pollbook/barcode-scanner-daemon/src/unitech_ts100_daemon.rs

```diff
@@ -0,0 +1,270 @@
+                    }
+                }
+            }
+        });
+    }
+
+    let port = match init_port(PORT_NAME, BAUD_RATE) {
+        Ok(p) => p,
+        Err(e) => {
+            log!(
+                event_id: EventId::DeviceAttached,
+                message: format!("Failed to connect to USB barcode scanner: {e}"),
+                event_type: EventType::SystemStatus,
+                disposition: Disposition::Failure
+            );
+
+            // Exit and allow systemctl config to restart daemon
+            log!(
+                EventId::ProcessTerminated;
+                EventType::SystemAction
+            );
+            exit(1);
+        }
+    };
+
+    log!(
+        event_id: EventId::DeviceAttached,
+        message: format!("Listening for barcode scan data on {}...", PORT_NAME),
+        event_type: EventType::SystemStatus,
+        disposition: Disposition::Success
+    );
+
+    let mut reader = BufReader::new(port);
+    let running = Arc::new(AtomicBool::new(true));
+
+    if let Err(e) = ctrlc::set_handler({
+        let running = running.clone();
+        move || {
+            running.store(false, Ordering::SeqCst);
+        }
+    }) {
+        log!(
+            event_id: EventId::ErrorSettingSigintHandler,
+            message: e.to_string(),
+            event_type: EventType::SystemStatus,
+            disposition: Disposition::Failure
+        );
+    }
+
+    let mut buf = Vec::new();
+    let mut start = Instant::now();
+
+    // Poll for barcode scan data
+    loop {
+        if !running.load(Ordering::SeqCst) {
+            break;
+        }
+
+        if start.elapsed() > HEARTBEAT_LOG_INTERVAL {
+            start = Instant::now();
+            log!(
+                EventId::Heartbeat;
+                EventType::SystemStatus
+            );
+        }
+
+        buf.clear();
+
+        match reader.read_until(b'\r', &mut buf) {
+            Ok(_) => {
+                // Scanner is configured by default to terminate data with '\r'
+                if buf.ends_with(&[b'\r']) {
+                    buf.pop();
+                }
+
+                if buf.len() == 0 {
+                    continue;
+                }
+
+                // We expect the input sequence to start with the compliance indicator
+                // so we just ignore it and continue to read the next line
+                if buf.as_slice() == COMPLIANCE_INDICATOR {
+                    continue;
+                }
+
+                match from_utf8(&buf) {
+                    Ok(s) => {
+                        match AamvaDocument::try_from(s) {
+                            Ok(document) => {
+                                let serialized = serde_json::to_string_pretty(&document)
+                                    .context("Failed to serialize AAMVA document to JSON")?;
+                                let mut guard = clients.lock().unwrap();
```

Comment:
If you end up going with `nusb` and switching this code to `async`, consider using `tokio::sync::Mutex` so you can call `clients.lock().await`. In the case where this lock tries to run while a client is connecting one of them may end up trying to `unwrap` a `PoisonError`.

Reply:
Ended up removing support for multiple clients but we may want to revisit this if `systemctl` configs aren't enough to reliably reconnect UDS server/client in the event of daemon and/or node backend crash.